lsp/server/src/server.ts

```diff
@@ -78,56 +96,24 @@ connection.onInitialized(() => {
 			connection.console.log('Workspace folder change event received.');
 		});
 	}
-});
-
-// The example settings
-interface ExampleSettings {
-	maxNumberOfProblems: number;
-}
-
-// Cache the settings of all open documents
-const documentSettings: Map<string, Thenable<ExampleSettings>> = new Map();
-
-connection.onDidChangeConfiguration(change => {
-	if (hasConfigurationCapability) {
-		// Reset all cached document settings
-		documentSettings.clear();
-		if (change.settings.SA4U){
-			messDef = change.settings.SA4U.messageDefinition;
-			documents.all().forEach(validateTextDocument);
+	connection.workspace.getWorkspaceFolders().then((folders:null|WorkspaceFolder[]|void) => {
+		if (folders) {
+			folders.forEach((folder) => {
+				dockerContainer(folder);
+			});
 		}
-	}
+	});
 });
 
-// Only keep settings for open documents
-documents.onDidClose(e => {
-	documentSettings.delete(e.document.uri);
-});
-
-// The content of a text document has changed. This event is emitted
-// when the text document first opened or when its content has changed.
-documents.onDidSave(change => {
-	validateTextDocument(change.document);
-});
-
-// Validate documents whenever they're opened.
-documents.onDidOpen(e => {
-	validateTextDocument(e.document);
-});
-
-async function validateTextDocument(textDocument: TextDocument): Promise<void> {
-	let path = decodeURIComponent(textDocument.uri);
-	path = path.substring(0,path.lastIndexOf("/")+1);
+async function dockerContainer (folder: WorkspaceFolder) {
+	let path = decodeURIComponent(folder.uri);
+	const filePath = path;
 	path = path.replace(/(^\w+:|^)\/\//, '');
 	path = path.replace(/:/, '');
-	const diagnostics: Diagnostic[] = [];
-	const execPromise = promisify(exec);
-	//const sa4uPromise = execPromise(`docker container run --rm -v ${path}:/src/ sa4u -c /src/ -p /src/ex_prior.json -m /src/CMASI.xml`);
+	const diagnosticsMap = new Map<string, Diagnostic[]>();
 	try {
-		const sa4uZ3Promise = execPromise(`docker container run --rm --mount type=bind,source="${path}",target="/src/" sa4u-z3 -c /src/ -p /src/ex_prior.json -m /src/${messDef}`);
-		//const sa4uOutput = await sa4uPromise;
-		const sa4uZ3Output = await sa4uZ3Promise;
-
+		// eslint-disable-next-line @typescript-eslint/no-var-requires
+		const readline = require('readline');
```

Comment:
nit: put all requires at the top of the file.
also, is there a way to import readline without having to disable eslint?

Reply:
I can't get the import to work.
https://stackoverflow.com/questions/52046008/why-cant-i-import-as-readline-from-readline-in-nodejs-v10